File: src/utils/decoder.ts
function decodeJWT(token: string): { header: object; payload: object } | null {
    const parts = token.split('.');
    if (parts.length !== 3) {
        return null; // Invalid JWT
    }

    const header = JSON.parse(Buffer.from(parts[0], 'base64').toString());
    const payload = JSON.parse(Buffer.from(parts[1], 'base64').toString());

    return { header, payload };
}

export { decodeJWT };

File: src/utils/decoder.py
import json
import base64
from typing import Dict, Optional, Any

def decode_jwt(token: str) -> Optional[Dict[str, Any]]:
    #Decode JWT token and return header and payload.
    parts = token.split('.')
    if len(parts) != 3:
        return None  # Invalid JWT
    
    try:
        # Add padding if needed
        header_b64 = parts[0] + '=' * (4 - len(parts[0]) % 4)
        payload_b64 = parts[1] + '=' * (4 - len(parts[1]) % 4)
        
        header = json.loads(base64.b64decode(header_b64))
        payload = json.loads(base64.b64decode(payload_b64))
        
        return {"header": header, "payload": payload}
    except Exception:
        return None